src/hotspot/share/classfile/protectionDomainCache.cpp

```diff
@@ -128,108 +159,69 @@ void ProtectionDomainCacheTable::unlink() {
 
   // Purge any deleted entries outside of the SystemDictionary_lock.
   purge_deleted_entries();
+  int oops_removed = purge_entries_from_table();
 
-  MutexLocker ml(SystemDictionary_lock);
-  int oops_removed = 0;
-  for (int i = 0; i < table_size(); ++i) {
-    ProtectionDomainCacheEntry** p = bucket_addr(i);
-    ProtectionDomainCacheEntry* entry = bucket(i);
-    while (entry != NULL) {
-      oop pd = entry->object_no_keepalive();
-      if (pd != NULL) {
-        p = entry->next_addr();
-      } else {
-        oops_removed++;
-        LogTarget(Debug, protectiondomain, table) lt;
-        if (lt.is_enabled()) {
-          LogStream ls(lt);
-          ls.print_cr("protection domain unlinked at %d", i);
-        }
-        entry->literal().release(Universe::vm_weak());
-        *p = entry->next();
-        free_entry(entry);
-      }
-      entry = *p;
-    }
-  }
   _total_oops_removed += oops_removed;
   _dead_entries = false;
 }
 
-void ProtectionDomainCacheTable::print_on(outputStream* st) const {
+void ProtectionDomainCacheTable::print_on(outputStream* st) {
```

Comment:
It is a little disconcerting that `print_on` can no longer be a `const` function!

Reply:
It's static, so it can't.

Reply:
Ah I see - I hadn't noticed it was now static.